src/workers/parquet-exporter.ts

```diff
@@ -15,92 +15,157 @@
     this.duckDbPath = duckDbPath;
+    this.bundlesDbPath = bundlesDbPath;
+    this.coreDbPath = coreDbPath;
   }
 
-  static async create({
-    log,
-    duckDbPath,
-    sqliteBundlesDbPath,
-    sqliteCoreDbPath,
+  async export({
+    outputDir,
+    startHeight,
+    endHeight,
+    maxFileRows,
   }: {
-    log: winston.Logger;
-    duckDbPath: string;
-    sqliteBundlesDbPath: string;
-    sqliteCoreDbPath: string;
-  }) {
-    const logger = log.child({ class: 'ParquetExporter' });
-    let db: Database;
+    outputDir: string;
+    startHeight: number;
+    endHeight: number;
+    maxFileRows: number;
+  }): Promise<void> {
+    if (this.isExporting) {
+      this.log.error('An export is already in progress');
+      return;
+    }
+    this.isExporting = true;
 
-    try {
-      db = await Database.create(duckDbPath);
+    return new Promise((resolve, reject) => {
+      const workerUrl = new URL('./parquet-exporter.js', import.meta.url);
+      this.worker = new Worker(workerUrl, {
+        workerData: {
+          log: this.log,
+          outputDir,
+          startHeight,
+          endHeight,
+          maxFileRows,
+          duckDbPath: this.duckDbPath,
+          bundlesDbPath: this.bundlesDbPath,
+          coreDbPath: this.coreDbPath,
+        },
+      });
 
-      const duckDbSchema = readFileSync(
-        './src/database/duckdb/schema.sql',
-        'utf8',
-      );
-      await db.exec(duckDbSchema);
+      let startTime: number;
 
-      logger.debug('DuckDB schema created');
-    } catch (error) {
-      logger.error('Error creating DuckDB database', error);
-      throw error;
-    }
+      this.worker.on('online', () => {
+        startTime = Date.now();
 
-    try {
-      await db.exec(`INSTALL sqlite; LOAD sqlite;`);
-      await db.exec(
-        `ATTACH '${sqliteBundlesDbPath}' AS sqlite_bundles_db (TYPE sqlite);`,
-      );
-      await db.exec(
-        `ATTACH '${sqliteCoreDbPath}' AS sqlite_core_db (TYPE sqlite);`,
-      );
+        this.log.info('Started Parquet export', {
+          outputDir,
+          startHeight,
+          endHeight,
+          maxFileRows,
+        });
 
-      logger.debug('SQLite databases loaded');
-    } catch (error) {
-      logger.error('Error loading SQLite databases', error);
-      throw error;
-    }
+        this.worker?.postMessage('start');
+      });
 
-    logger.info('DuckDB database created!');
+      this.worker.on('message', (message: any) => {
+        if (message.eventName === EXPORT_COMPLETE) {
+          const endTime = Date.now();
+          const duration = (endTime - startTime) / 1000; // Convert to seconds
 
-    return new ParquetExporter({
-      log: logger,
-      db,
-      duckDbPath,
+          this.log.info(
+            `Parquet export completed in ${duration.toFixed(2)} seconds`,
+            { outputDir, startHeight, endHeight, maxFileRows },
+          );
+
+          this.isExporting = false;
+          this.worker?.terminate();
```

Comment:
Worker is terminated everytime it finishes or fail a job and a new worker is created whenever we call `export`